Anteproyecto/README.md

```diff
@@ -1,123 +1,251 @@
+
+![alt text](<img/circuito.png>)
+
+*Imagen 5: Diagrama en bloques general*
+
+#### Circuito de acondicionamiento ACS712
+En la *Imagen 6* se ve el diagrama en bloques del sensor ACS712. Como se observa, la entrada de corriente se mide con un shunt y está aislada de la salida.
+
+![alt text](<img/sens_corr_datasheet.png>)
+
+*Imagen 6: Imagen del sensor y diagrama en bloques del circuito integrado*
+
+Un punto a tener en cuenta es que la señal viene acondicionada entre 0 y 5 V. Por eso hay que re-acondicionarlo para ajustarle el rango dinámico a 0 - 3.3 V.
+
+Por otro lado, hay que añadirle un filtro para evitar aliasing al momento de muestrear en el microcontrolador. En este caso la frecuencia de corte es de 5 kHz, es decir, vamos a leer hasta el armónico 100.
+
+![alt text](<img/acond_sens_corr.png>)
+
+*Imagen 7: Acondicionamiento sensor de corriente*
+
+En la *Imagen 8* e *Imagen 9* se pueden ver las simulaciones en tiempo y en frecuencia para una señal de corriente de 5A.
+
+![alt text](<img/acond_sens_corr_t.png>)
+
+*Imagen 8: Respuesta temporal para entrada de 5A*
+
+![alt text](<img/acond_sens_corr_f.png>)
+
+*Imagen 9: Respuesta en frecuencia para entrada de 5A*
+
+#### Circuito Detector de Ciclos
+Este circuito entrega 3.3 V cuando llega cuando la tensión de línea está en el semi-ciclo positivo y 0 V cuando está en el negativo. En la *Imagen 10* se ve el circuito y en la *Imagen 11* las señales de entrada (línea) y salida (tensión lógica).
+Cabe destacar que la lógica está aislada de la línea mediante un opto-acoplador.
+
+![alt text](<img/detect_ciclos.png>)
+
+*Imagen 10: Circuito de detección de ciclos*
+
+![alt text](<img/detect_ciclos_t.png>)
 
-![alt text](<img/det_ciclos.jpg>)
+*Imagen 11: Formas de onda de entrada (tensión de línea) y salida (tensión lógica)*
 
-### Double buffering en la entrada y la salida
-Tanto la entrada como la salida tendrán un esquema de ping pong buffering, es decir, existirán los punteros
+#### Circuito Display
+Para el display se utiliza el esquema de la *Imagen 12*
+
+![alt text](<img/display.png>)
+
+*Imagen 12: Display*
+
+#### Circuito DAC
+Para el DAC simplemente voy a utilizar un circuito RC para suavizar las altas frecuencias del DAC (*Imagen 13*)
+
+![alt text](<img/dac.png>)
+
+*Imagen 13: DAC*
+
+Cabe destacar que esta salida es provisoria. En el proyecto final la salida no será DAC sino que será PWM y controlarán a la etapa de potencia que se encarga de transformar la señal PWM a corriente consumida en la línea.
+
+### Adquisición de datos
+Para adquirir los datos de manera eficiente propongo la implementación de la técnica de "double buffering en la entrada y la salida"<sup>[6]</sup>.
+Para el procesamiento en tiempo real es una buena estrategia para hacer uso eficiente del procesador. La idea es que el procesador no use demasiados recursos en samplear, queremos que se aproveche lo máximo posible para resolver las operaciones matemáticas requeridas en el procesamiento.
```

Comment:
prefiero que uses "muestrear" o mejor aún "tomar muestras" en vez de "samplear", que no existe.

Respecto al double-buffering, no se hace por performance sino para no perder datos... cuando terminaste de llenar el buffer ping, no tenés dónde meter(te) los datos que llegan hasta que lo vaciás, luego disponés de un segundo buffer, pong, y conmutás. Esto, a la vez, te permite, cuidadosamente planificado, evitar la copia de los datos (pero sólo si te alcanza el tiempo) y procesarlos in-place, lo cual te mejora la performance, pero A no implica B. En el caso que no te alcance el tiempo, se usa un ring buffer, el ping-pong es un caso especial del ring, un anillo de 2 buffers.